src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,883 @@
+                        partitionId: NotApplicable,
+                        details: $"Unexpected error occurred while trying to manage table partition leases: {exception}");
+                    
+                    consecutiveFailureCount++;
+                }
+
+                // If table update failed, we re-read the table immediately to obtain the latest ETag.
+                // In the case of too many successive failures, we wait before retrying to prevent excessive logs.
+                if (consecutiveFailureCount > 0 && consecutiveFailureCount < MaxFailureCount)
+                {
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
+
+                try
+                {
+                    if (isShuttingDown || forcefulShutdownToken.IsCancellationRequested)
+                    {
+                        // If shutdown is required, we sleep for a short period to ensure a relatively fast shutdown process
+                        await Task.Delay(timeToSleep, forcefulShutdownToken);
+                    }
+                    else
+                    {
+                        // Normal case: the amount of time we sleep varies depending on the situation.
+                        await Task.Delay(timeToSleep, gracefulShutdownToken);
+                    }
+                }
+                catch (OperationCanceledException) when (gracefulShutdownToken.IsCancellationRequested)
+                {
+                    // Shutdown requested, but we still need to release all leases
+                    if (!isShuttingDown)
+                    {
+                        isShuttingDown = true;
+                        this.settings.Logger.PartitionManagerInfo(
+                            this.storageAccountName,
+                            this.settings.TaskHubName,
+                            this.settings.WorkerId,
+                            partitionId: NotApplicable,
+                            details: $"Requested to cancel partition manager table manage loop. Initiate shutdown process.");
+                    }
+                }
+            }
+
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                partitionId: NotApplicable,
+                "Stopped background table partition manager loop.");
+        }
+
+        /// <summary>
+        /// Stop the partition manager. It first stops the partition manager loop to stop involving in balance in case it steals others' partitions in shutdown process.
+        /// And then start the Task ShutDown() to release all ownership leases. 
+        /// Worker will retry updating the table if the update was failed or any other exceptions occurred.
+        /// If the failure operations happened too many times, the loop waiting time will be extended to avoid excessive loggings.
+        /// </summary>
```

Comment:
nit: minor phrasing suggestions
```suggestion
        /// <summary>
        /// Stop the partition manager. It first stops the partition manager loop to prevent stealing partitions during shutdown.
        /// Then it starts the Task ShutDown() to release all ownership leases. 
        /// Worker will retry updating the table if the update failed or if any other exceptions occurred.
        /// In the case of too many failed operations, the loop waiting time will be extended to avoid excessive logs.
        /// </summary>
```